segmenter/video_segmenter.go

```diff
@@ -55,19 +52,18 @@ type FFMpegVideoSegmenter struct {
 	WorkDir        string
 	LocalRtmpUrl   string
 	StrmID         string
-	ffmpegPath     string
 	curSegment     int
 	curPlaylist    *m3u8.MediaPlaylist
 	curPlWaitTime  time.Duration
 	curSegWaitTime time.Duration
 	SegLen         time.Duration
```

Comment:
Looks like the SegLen field is also used within `pollPlaylist` , so we probably should keep that field in there for now. We can do either:

1. Remove `SegLen` from the `NewFFMpegVideoSegmenter` params and keep it as an opt to RTMPToHLS. Then in RTMPToHLS, set `s.SegLen = opt.SegLength`. Since it's an opt, we probably should also set a default value if unset.

2. Remove `SegmenterOptions.SegLength` and keep it as a mandatory parameter in NewFFMpegVideoSegmenter.

3. A combination of 1 & 2 by passing SegmenterOptions into NewFFMpegVideoSegmenter and removing the opt from RTMPToHLS.

I slightly prefer the third option. The first option leads to a potentially unsafe API since it makes PollPlaylist dependent RTMPToHLS being called first. Not too fond of the second option, since In general I'm in favor of 'sane defaults' as it leads to an API that's easier to get started with. That means documenting the option somewhere though.

Reply:
Yeah I also think option 3 makes the most sense.

Reply:
Option 3 implemented in https://github.com/livepeer/lpms/pull/48/commits/d0866ac11e3bd743e53b8a77bd3e07b847456a47 , although it's mixed up with the 3 second retry thing. Can split up if you'd like.